node/tests/reconstruct_test.go

```diff
@@ -0,0 +1,120 @@
+//go:build !race
+
+package tests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/celestiaorg/celestia-node/ipld"
+
+	"github.com/celestiaorg/celestia-node/service/share"
+
+	"github.com/celestiaorg/celestia-node/node"
+	"github.com/celestiaorg/celestia-node/node/tests/swamp"
+)
+
```

Comment:
Missing description of the test
```suggestion
/*
Test-Case: Full Node reconstructs blocks from a Bridge node
Pre-Reqs: 
- First 20 blocks have a block size of 16
- Blocktime is 100 ms
Steps:
1. Create a Bridge Node(BN)
2. Start a BN
3. Create a Full Node(FN) with BN as a trusted peer
4. Start a FN 
5. Check that a FN can retrieve shares from 1 to 20 blocks
*/
```